Build failed. Error cannot find symbol builder.setNavigationBarColor

```
> Task :react-native-inappbrowser-reborn:compileReleaseJavaWithJavac FAILED
/Users/alexandrsmirnov/Work/E-Orda/Colibri-mobile/node_modules/react-native-inappbrowser-reborn/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java:107: error: cannot find symbol
        builder.setNavigationBarColor(Color.parseColor(colorString));
               ^
  symbol:   method setNavigationBarColor(int)
  location: variable builder of type Builder
/Users/alexandrsmirnov/Work/E-Orda/Colibri-mobile/node_modules/react-native-inappbrowser-reborn/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java:116: error: cannot find symbol
        builder.setNavigationBarDividerColor(Color.parseColor(colorString));
               ^
  symbol:   method setNavigationBarDividerColor(int)
  location: variable builder of type Builder
2 errors
```
### Which platform(s) does your issue occur on?
- Android 

**Package.json**
```
"react-native": "^0.64.2"  
"react-native-inappbrowser-reborn": "^3.6.1" 
```

> Task :react-native-inappbrowser-reborn:compileReleaseJavaWithJavac FAILED /Users/alexandrsmirnov/Work/E-Orda/Colibri-mobile/node_modules/react-native-inappbrowser-reborn/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java:107: error: cannot find symbol builder.setNavigationBarColor(Color.parseColor(colorString)); ^ symbol: method setNavigationBarColor(int) location: variable builder of type Builder /Users/alexandrsmirnov/Work/E-Orda/Colibri-mobile/node_modules/react-native-inappbrowser-reborn/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java:116: error: cannot find symbol builder.setNavigationBarDividerColor(Color.parseColor(colorString)); ^ symbol: method setNavigationBarDividerColor(int) location: variable builder of type Builder 2 errors

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,9 @@ in case of vulnerabilities.
 
 ## [Unreleased]
 
+### Fixed
+- Fix `Build failed. Error cannot find symbol builder.setNavigationBarColor` error for Android Support ([#281](https://github.com/proyecto26/react-native-inappbrowser/pull/281)).
+
 ## [3.6.1] - 2021-06-27
 
 ### Added

--- a/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
+++ b/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
@@ -64,11 +64,10 @@ public class RNInAppBrowser {
   private static final Pattern animationIdentifierPattern = Pattern.compile("^.+:.+/");
 
   public Integer setColor(CustomTabsIntent.Builder builder, final ReadableMap options, String key, String method, String colorName) {
-    String colorString = "";
+    String colorString = options.getString(key);
     Integer color = null;
     try {
-      if (options.hasKey(key)) {
-        colorString = options.getString(key);
+      if (colorString != null) {
         color = Color.parseColor(colorString);
         Method findMethod = builder.getClass().getDeclaredMethod(method, int.class);
         findMethod.invoke(builder, color);

--- a/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
+++ b/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
@@ -63,30 +63,23 @@ public class RNInAppBrowser {
   private Activity currentActivity;
   private static final Pattern animationIdentifierPattern = Pattern.compile("^.+:.+/");
 
-  public void setNavigationColors(CustomTabsIntent.Builder builder, final ReadableMap options) {
-    if (options.hasKey(KEY_NAVIGATION_BAR_COLOR)) {
-      final String colorString = options.getString(KEY_NAVIGATION_BAR_COLOR);
-      try {
-        Method setNavigationBarColor = builder.getClass().getDeclaredMethod("setNavigationBarColor", int.class);
-        setNavigationBarColor.invoke(builder, Color.parseColor(colorString));
-      } catch (Exception e) {
-        if (e instanceof IllegalArgumentException) {
-          throw new JSApplicationIllegalArgumentException(
-                  "Invalid navigation bar color '" + colorString + "': " + e.getMessage());
-        }
+  public Integer setColor(CustomTabsIntent.Builder builder, final ReadableMap options, String key, String method, String colorName) {
+    String colorString = "";
+    Integer color = null;
+    try {
+      if (options.hasKey(key)) {
+        colorString = options.getString(key);
+        color = Color.parseColor(colorString);
+        Method findMethod = builder.getClass().getDeclaredMethod(method, int.class);
+        findMethod.invoke(builder, color);
       }
-    }
-    if (options.hasKey(KEY_NAVIGATION_BAR_DIVIDER_COLOR)) {
-      final String colorString = options.getString(KEY_NAVIGATION_BAR_DIVIDER_COLOR);
-      try {
-        Method setNavigationBarDividerColor = builder.getClass().getDeclaredMethod("setNavigationBarDividerColor", int.class);
-        setNavigationBarDividerColor.invoke(builder, Color.parseColor(colorString));
-      } catch (Exception e) {
-        if (e instanceof IllegalArgumentException) {
-          throw new JSApplicationIllegalArgumentException(
-                "Invalid navigation bar divider color '" + colorString + "': " + e.getMessage());
-        }
+    } catch (Exception e) {
+      if (e instanceof IllegalArgumentException) {
+        throw new JSApplicationIllegalArgumentException(
+                "Invalid " + colorName + " color '" + colorString + "': " + e.getMessage());
       }
+    } finally {
+      return color;
     }
   }
 
@@ -110,26 +103,13 @@ public class RNInAppBrowser {
 
     CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
     isLightTheme = false;
-    if (options.hasKey(KEY_TOOLBAR_COLOR)) {
-      final String colorString = options.getString(KEY_TOOLBAR_COLOR);
-      try {
-        builder.setToolbarColor(Color.parseColor(colorString));
-        isLightTheme = toolbarIsLight(colorString);
-      } catch (IllegalArgumentException e) {
-        throw new JSApplicationIllegalArgumentException(
-                "Invalid toolbar color '" + colorString + "': " + e.getMessage());
-      }
+    final Integer toolBarColor = setColor(builder, options, KEY_TOOLBAR_COLOR, "setToolbarColor", "toolbar");
+    if (toolBarColor != null) {
+      isLightTheme = toolbarIsLight(toolBarColor);
     }
-    if (options.hasKey(KEY_SECONDARY_TOOLBAR_COLOR)) {
-      final String colorString = options.getString(KEY_SECONDARY_TOOLBAR_COLOR);
-      try {
-        builder.setSecondaryToolbarColor(Color.parseColor(colorString));
-      } catch (IllegalArgumentException e) {
-        throw new JSApplicationIllegalArgumentException(
-                "Invalid secondary toolbar color '" + colorString + "': " + e.getMessage());
-      }
-    }
-    setNavigationColors(builder, options);
+    setColor(builder, options, KEY_SECONDARY_TOOLBAR_COLOR, "setSecondaryToolbarColor", "secondary toolbar");
+    setColor(builder, options, KEY_NAVIGATION_BAR_COLOR, "setNavigationBarColor", "navigation bar");
+    setColor(builder, options, KEY_NAVIGATION_BAR_DIVIDER_COLOR, "setNavigationBarDividerColor", "navigation bar divider");
 
     if (options.hasKey(KEY_DEFAULT_SHARE_MENU_ITEM) && 
         options.getBoolean(KEY_DEFAULT_SHARE_MENU_ITEM)) {
@@ -304,8 +284,8 @@ public class RNInAppBrowser {
     }
   }
 
-  private Boolean toolbarIsLight(String themeColor) {
-    return ColorUtils.calculateLuminance(Color.parseColor(themeColor)) > 0.5;
+  private Boolean toolbarIsLight(int themeColor) {
+    return ColorUtils.calculateLuminance(themeColor) > 0.5;
   }
 
   private List<ResolveInfo> getPreferredPackages(Context context) {

--- a/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
+++ b/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
@@ -103,9 +103,9 @@ public class RNInAppBrowser {
 
     CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
     isLightTheme = false;
-    final Integer toolBarColor = setColor(builder, options, KEY_TOOLBAR_COLOR, "setToolbarColor", "toolbar");
-    if (toolBarColor != null) {
-      isLightTheme = toolbarIsLight(toolBarColor);
+    final Integer toolbarColor = setColor(builder, options, KEY_TOOLBAR_COLOR, "setToolbarColor", "toolbar");
+    if (toolbarColor != null) {
+      isLightTheme = toolbarIsLight(toolbarColor);
     }
     setColor(builder, options, KEY_SECONDARY_TOOLBAR_COLOR, "setSecondaryToolbarColor", "secondary toolbar");
     setColor(builder, options, KEY_NAVIGATION_BAR_COLOR, "setNavigationBarColor", "navigation bar");

--- a/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
+++ b/android/src/main/java/com/proyecto26/inappbrowser/RNInAppBrowser.java
@@ -27,6 +27,7 @@ import com.facebook.react.bridge.JSApplicationIllegalArgumentException;
 import org.greenrobot.eventbus.EventBus;
 import org.greenrobot.eventbus.Subscribe;
 
+import java.lang.reflect.Method;
 import java.util.Arrays;
 import java.util.regex.Pattern;
 import java.util.List;
@@ -61,6 +62,33 @@ public class RNInAppBrowser {
   private Boolean isLightTheme;
   private Activity currentActivity;
   private static final Pattern animationIdentifierPattern = Pattern.compile("^.+:.+/");
+
+  public void setNavigationColors(CustomTabsIntent.Builder builder, final ReadableMap options) {
+    if (options.hasKey(KEY_NAVIGATION_BAR_COLOR)) {
+      final String colorString = options.getString(KEY_NAVIGATION_BAR_COLOR);
+      try {
+        Method setNavigationBarColor = builder.getClass().getDeclaredMethod("setNavigationBarColor", int.class);
+        setNavigationBarColor.invoke(builder, Color.parseColor(colorString));
+      } catch (Exception e) {
+        if (e instanceof IllegalArgumentException) {
+          throw new JSApplicationIllegalArgumentException(
+                  "Invalid navigation bar color '" + colorString + "': " + e.getMessage());
+        }
+      }
+    }
+    if (options.hasKey(KEY_NAVIGATION_BAR_DIVIDER_COLOR)) {
+      final String colorString = options.getString(KEY_NAVIGATION_BAR_DIVIDER_COLOR);
+      try {
+        Method setNavigationBarDividerColor = builder.getClass().getDeclaredMethod("setNavigationBarDividerColor", int.class);
+        setNavigationBarDividerColor.invoke(builder, Color.parseColor(colorString));
+      } catch (Exception e) {
+        if (e instanceof IllegalArgumentException) {
+          throw new JSApplicationIllegalArgumentException(
+                "Invalid navigation bar divider color '" + colorString + "': " + e.getMessage());
+        }
+      }
+    }
+  }
 
   public void open(Context context, final ReadableMap options, final Promise promise, Activity activity) {
     final String url = options.getString("url");
@@ -101,24 +129,8 @@ public class RNInAppBrowser {
                 "Invalid secondary toolbar color '" + colorString + "': " + e.getMessage());
       }
     }
-    if (options.hasKey(KEY_NAVIGATION_BAR_COLOR)) {
-      final String colorString = options.getString(KEY_NAVIGATION_BAR_COLOR);
-      try {
-        builder.setNavigationBarColor(Color.parseColor(colorString));
-      } catch (IllegalArgumentException e) {
-        throw new JSApplicationIllegalArgumentException(
-                "Invalid navigation bar color '" + colorString + "': " + e.getMessage());
-      }
-    }
-    if (options.hasKey(KEY_NAVIGATION_BAR_DIVIDER_COLOR)) {
-      final String colorString = options.getString(KEY_NAVIGATION_BAR_DIVIDER_COLOR);
-      try {
-        builder.setNavigationBarDividerColor(Color.parseColor(colorString));
-      } catch (IllegalArgumentException e) {
-        throw new JSApplicationIllegalArgumentException(
-                "Invalid navigation bar divider color '" + colorString + "': " + e.getMessage());
-      }
-    }
+    setNavigationColors(builder, options);
+
     if (options.hasKey(KEY_DEFAULT_SHARE_MENU_ITEM) && 
         options.getBoolean(KEY_DEFAULT_SHARE_MENU_ITEM)) {
       builder.addDefaultShareMenuItem();

